src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Usage/ProvideCorrectArgumentToEnumHasFlag.cs

```diff
@@ -54,26 +43,18 @@ public override void Initialize(AnalysisContext context)
                 {
                     var invocation = (IInvocationOperation)context.Operation;
 
-                    if (invocation.TargetMethod.ContainingType.SpecialType != SpecialType.System_Enum ||
-                        invocation.Arguments.Length != 1 ||
-                        invocation.Instance == null ||
-                        invocation.TargetMethod.Name != "HasFlag" ||
-                        invocation.Arguments[0].Value is not IConversionOperation conversion ||
-                        invocation.Instance.Type == null ||
-                        invocation.Instance.Type.TypeKind == TypeKind.TypeParameter ||
-                        conversion.Operand.Type?.TypeKind == TypeKind.TypeParameter)
-                    {
-                        return;
-                    }
-
-                    if (!invocation.Instance.Type.Equals(conversion.Operand.Type))
+                    if (invocation.TargetMethod.ContainingType.SpecialType == SpecialType.System_Enum &&
```

Comment:
💭 I think I would have preferred to not change the flow control layout here, but I suppose what's done is done

Reply:
I can revert, it just seemed weird to have all the conditions for the return and then one if for the actual condition.